### PR TITLE
Add support for dotnet build

### LINKED ISSUE #17: Support dotnet execution tool
Is there any chance to get support in StrongNamer to be executed in dotnet core CLI?

I tried to follow this tutorial to convert it, but without luck.

If you want I can pass you the WIP.

### PR DESCRIPTION
**Pull request by clairernovotny**
Fixes #17

This PR adds support for building with `dotnet` on CoreCLR. It uses Cecil 0.11, which added support for that. 

It also converts all the projects to SDK-style, adds an Azure Pipelines yaml and auto versioning.

### CODE REVIEW

**Comment by onovotny** [comment]
Progress, builds/doesn't error out, but doesn't work yet. Investigating.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+trigger:
```
Comment (dsplaisted) [inline azure-pipelines.yml]: Did you hook this up to a pipeline anywhere and test it?  How do you expect me to hook it up to CI?

File: version.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "version": "0.1",
```
Comment (dsplaisted) [inline version.json]: What will the versions end up looking like here?

Comment (onovotny) [reply version.json]: They'll be 0.1.{height} since last increment.

File: azure-pipelines.yml (hunk 1 above)
Comment (onovotny) [reply azure-pipelines.yml]: Setup a free Azure DevOps pipeline? :)

I haven't tested it but I'm mostly sure it works ;) It's based on many others I've done.

File: src/StrongNamer/AddStrongName.cs [hunk 3]
```diff
@@ -217,5 +217,40 @@ private static byte[] GetKeyTokenFromKey(byte[] fullKey)
 
             return hash.Reverse().Take(8).ToArray();
         }
+
+        // From Cecil
+        // https://github.com/jbevain/cecil/pull/548/files
```
[the remark this answers is not in the document]
Comment (onovotny) [reply src/StrongNamer/AddStrongName.cs]: Correct. I've suggested that but it's not likely to be soon. Fody has its own private copy of StrongNameKeyPair.

[the remark this answers is not in the document]
Comment (onovotny) [reply src/StrongNamer/AddStrongName.cs]: https://github.com/jbevain/cecil/pull/548#issuecomment-549185165

File: version.json (hunk 2 above)
Comment (dsplaisted) [reply version.json]: Does that mean that the NuGet package versions will also include the commit height?  I'd prefer to avoid that.

Comment (onovotny) [reply version.json]: Up to you -- I've found that no one seems to care about the patch version; you can see it in action in any of my libraries (rx/ix, humanizer, etc). It's a way to get a reproducible build/version number that auto versions. Driven by commit depth has been the most reliable way.

I can remove auto versioning if you have another approach you prefer.

**Comment by onovotny** [comment]
You can see it working by looking at the commits/builds here:
https://github.com/reactiveui/refit/commits/master

File: src/StrongNamer/AddStrongName.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (jzabroski) [reply src/StrongNamer/AddStrongName.cs:222]: Why wouldn't this logic just live in Microsoft.CodeAnalysis?  There is already a DesktopStrongNameProvider.

File: version.json (hunk 2 above)
Comment (jzabroski) [reply version.json:2]: I like GitVersion personally.  I suppose you can drive it through GitInfo instead.

**Pull request merged**